utils/python/CIME/SystemTests/pem.py

```diff
@@ -12,119 +12,41 @@
-    def _pem_first_phase(self):
-
-        # Reset beginning test settings
-        expect(os.path.isfile("env_mach_pes.xml.1"),
-               "ERROR: env_mach_pes.xml.1 does not exist, run case.build" )
-
-        shutil.copy("env_mach_pes.xml.1", "env_mach_pes.xml")
-        shutil.copy("env_mach_pes.xml.1", "LockedFiles/env_mach_pes.xml")
-
-        exeroot = self._case.get_value("EXEROOT")
-        cime_model = CIME.utils.get_model()
-        exefile  = "%s/%s.exe"%(exeroot,cime_model)
-        exefile1 = "%s/%s.exe.PEM1"%(exeroot,cime_model)
-        if (os.path.isfile(exefile)):
-            os.remove(exefile)
-        shutil.copy(exefile1, exefile)
-
         self._case.set_value("CONTINUE_RUN",False)
         self._case.set_value("REST_OPTION","none")
-        self._case.set_value("HIST_OPTION","$STOP_OPTION")
-        self._case.set_value("HIST_N","$STOP_N")
-        stop_n = self._case.get_value("STOP_N")
-        stop_option = self._case.get_value("STOP_OPTION")
-
-        self._case.flush()
-        logger.info("doing an %d %s initial test, no restarts written" % (stop_n, stop_option))
-
-        self.run_indv()
-
-    def _pem_second_phase(self):
 
-        expect(os.path.isfile("env_mach_pes.xml.2"),
-               "ERROR: env_mach_pes.xml.2 does not exist, run case.build" )
-
-        shutil.copy("env_mach_pes.xml.2", "env_mach_pes.xml")
-        shutil.copy("env_mach_pes.xml.2", "LockedFiles/env_mach_pes.xml")
-
-        exeroot = self._case.get_value("EXEROOT")
-        cime_model = CIME.utils.get_model()
-        exefile  = "%s/%s.exe"%(exeroot,cime_model)
-        exefile2 = "%s/%s.exe.PEM2"%(exeroot,cime_model)
-        if (os.path.isfile(exefile)):
-            os.remove(exefile)
-        shutil.copy(exefile2, exefile)
-
-        self._case.set_value("CONTINUE_RUN",False)
-        self._case.set_value("REST_OPTION","none")
-        self._case.set_value("HIST_OPTION","$STOP_OPTION")
-        self._case.set_value("HIST_N","$STOP_N")
-        stop_n = self._case.get_value("STOP_N")
-        stop_option = self._case.get_value("STOP_OPTION")
+    def _case_one_setup(self):
+        pass
 
-        self._case.flush()
-        logger.info("doing an %d %s initial test, no restarts written" % (stop_n, stop_option))
+    def _case_two_setup(self):
+        for comp in ['ATM','CPL','OCN','WAV','GLC','ICE','ROF','LND']:
+            ntasks = self._case.get_value("NTASKS_%s"%comp)
+            if ( ntasks > 1 ):
+                self._case.set_value("NTASKS_%s"%comp, ntasks/2)
+        case_setup(self._case, clean=False, test_mode=False, reset=True)
 
-        self.run_indv(suffix="modpes")
-        self._component_compare_test("base", "modpes")
+    def _config_run(self):
+        # These options must be set in the run phase,
+        # as their environment variables don't exist before then
+        for f in (self._activate_case1, self._activate_case2):
+            f()
+            self._case.set_value("HIST_OPTION","$STOP_OPTION")
```

Comment:
Actually I believe that the correct place for these settings is in cime_config/config_tests.xml  - that would be more consistent with other tests.